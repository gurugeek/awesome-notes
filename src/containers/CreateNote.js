import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, TextInput } from 'react-native';
import { back } from '../actions/navigation';
import { connect } from 'react-redux';
import uuid from 'uuid/v1';

import { saveNote } from '../actions/createNote';
import HeaderButton from '../components/common/HeaderButton';

class CreateNote extends Component {

    constructor(props) {
        super(props);
        this.state = {
            id: null,
            text: '',
            createdAt: null,
            favorite: false
        };

        this.saveNote = this.saveNote.bind(this);
    }

    static navigationOptions = ({ navigation }) => {
        const { state } = navigation;
        const cancel = state.params && state.params.cancel ? state.params.cancel : null;
        const save = state.params && state.params.save ? state.params.save : null;
        return {
            title: 'New Note',
            headerRight: <HeaderButton action={save} type='check' />,
            headerLeft: <HeaderButton action={cancel} type='cross' />
        }
    };

    componentDidMount() {
        this.props.navigation.setParams({
            cancel: this.props.cancel,
            save: this.saveNote
        });
    }

    saveNote() {

        this.setState({ id: uuid(), createdAt: new Date() }, () => this.props.save({}));
    }

    render() {
        return (
            <View style={{ flex: 1 }}>
                <TextInput
                    style={{ flex: 1, padding: 10 }}
                    multiline={true}
                    autoFocus={true}
                    placeholder='How do you plan to dominate the world?'
                    onChangeText={(text) => this.setState({ text })}
                    value={this.state.text}/>
            </View>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return ({
        cancel: () => {
            dispatch(back());
        },
        save: (note) => {
            dispatch(saveNote(note));
        }
    });
};

CreateNote.propTypes = {
    cancel: PropTypes.func.isRequired,
    save: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(CreateNote);