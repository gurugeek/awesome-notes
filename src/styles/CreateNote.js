export default {
    container: {
        flex: 1
    },
    textArea: {
        flex: 1,
        padding: 10
    }
}