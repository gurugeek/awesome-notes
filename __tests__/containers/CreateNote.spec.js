import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CreateNote from '../../src/containers/CreateNote';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore();

const mockNavigation = {
    setParams: () => {}
};

describe('<CreateNote /> container', () => {
    const store = mockStore();

    it('CreateNote should match with the snapshot', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <CreateNote navigation={mockNavigation} />
            </Provider>
        );
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });

    const createNoteShallow = shallow(
        <Provider store={store}>
            <CreateNote navigation={mockNavigation} />
        </Provider>
    );

    it('CreateNote should exists', () => {
        expect(createNoteShallow.length).toEqual(1);
    });
});