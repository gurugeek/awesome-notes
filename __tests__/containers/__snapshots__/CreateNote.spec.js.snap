// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateNote /> container CreateNote should match with the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoFocus={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="How do you plan to dominate the world?"
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 10,
          "textAlignVertical": "top",
        },
        Object {
          "marginBottom": 0,
        },
      ]
    }
    value=""
  />
</View>
`;
